controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -1045,6 +1055,34 @@ func (r *ReconcileHyperConverged) detectTaintedConfiguration(req *common.HcoRequ
 	}
 }
 
+func (r *ReconcileHyperConverged) getSystemHealthStatus(req *common.HcoRequest) string {
+	isConditionAvailableTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionAvailable)
+	isConditionDegradedTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionDegraded)
+	isConditionProgressingTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionProgressing)
+	isConditionReconcileCompleteTrue := req.Conditions.IsStatusConditionTrue(hcov1beta1.ConditionReconcileComplete)
+
+	isSystemHealthStatusError := !isConditionAvailableTrue || isConditionDegradedTrue
+	if isSystemHealthStatusError {
```

Comment:
Not sure about this logic. The result will be that we'll maybe get false negative, mostly during setup, until the system is fully function.

Reply:
Can you elaborate on what would be the reason for this false negative?
Is it related to the conditions' logic, or to the location in the reconciliation in which we update the system health?

Reply:
HyperConverged won't be available during setup. According to this code, we'll get an error metric in this case. Is that what we want? @sradco

Reply:
During setup, and only then, if I understand correctly, `ReconcileComplete=Unknown` ([hyperconverged_controller.go#L641](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/controllers/hyperconverged/hyperconverged_controller.go#L641)).
Would it be accurate to first check if `ReconcileComplete=Unknown`, and if so, to give the metric `warning`/`healthy`?

@sradco